materialize-bigquery/transactor.go

```diff
@@ -32,77 +35,86 @@ type transactor struct {
 	loggedStorageApiMessage bool
 }
 
-func newTransactor(
```

Comment:
Most of this diff is just from wrapping the `newTransactor` behavior in a closure so that it has access to the previously computed dialect, templates, and flag value.